taxonomy-manager-rest-server/src/main/java/com/digirati/taxman/rest/server/taxonomy/ProjectModelRepository.java

```diff
@@ -0,0 +1,58 @@
+package com.digirati.taxman.rest.server.taxonomy;
+
+import com.digirati.taxman.common.taxonomy.ProjectModel;
+import com.digirati.taxman.rest.server.taxonomy.mapper.ProjectMapper;
+import com.digirati.taxman.rest.server.taxonomy.storage.ProjectDao;
+import com.digirati.taxman.rest.server.taxonomy.storage.ProjectDataSet;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+
+@ApplicationScoped
+public class ProjectModelRepository {
+
+    @Inject
+    ProjectDao projectDao;
+
+    @Inject
+    ProjectMapper projectMapper;
+
+    /**
+     * Stores a new project in the database, along with mappings between the project and any associated concept schemes.
+     *
+     * @param project the project to create
+     * @return the created project
+     */
+    @Transactional(Transactional.TxType.REQUIRED)
+    public ProjectModel create(ProjectModel project) {
+        ProjectDataSet dataSet = projectMapper.map(project);
+        projectDao.storeDataSet(dataSet);
```

Comment:
This might not be the correct place to handle this, but are we doing any checks to ensure that a project with the provided name doesn't already exist, before we get this far? I can't see where such checks are happening...

Reply:
We're not - if the slug already exists it'll store it as an update to the existing project, which now that I think about it again is wrong. I'll add something here to check that

Reply:
> if the slug already exists it'll store it as an update to the existing project

Project creation comes in as a `POST`, update comes in as a `PUT`. You can't store it as an update if someone `POST`'s to the `/projects` endpoint using an existing slug, that can only occur if someone `PUT`'s to the `/project/{slug}` endpoint.

Reply:
The DAO uses the same method in either case. The only difference between the two endpoints once you get into them is what we return from them

Reply:
That's even more wrong, it just introduces unpredictable behaviour. `POST`'ing to `/projects` should **not** update the project if it already exists, only `PUT` to `/projects/{slug}` should do that.

The behaviour exposed by each endpoint must be tightly controlled, it shouldn't (read: mustn't) leak like that.

Reply:
This same issue affects concept schemes too. I'll fix it for projects as part of this change then make another one fixing it for concept schemes

Reply:
Actually, scratch that. This exact issue doesn't affect concept schemes because we generate the UUID as part of the creation, although a related issue exists whereby the `PUT` can create a new scheme. The fix to that should be to 404 if the scheme being updated doesn't exist - which requires the new `EmptyResultExceptionMapper` class from this change plus a single line added to the `ConceptSchemeModelRepository` - so I'll fix that as part of this change